3dText_server.lua

```diff
@@ -0,0 +1,164 @@
+--SERVER
+
+TextsList = {}
+
+local random = math.random
+local function uuid()
+    local template ='xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'
+    return string.gsub(template, '[xy]', function (c)
+        local v = (c == 'x') and random(0, 0xf) or random(8, 0xb)
+        return string.format('%x', v)
+    end)
+end
+
+--Creates 3D text with specified parameters and assigns unique GUID
+function create3dText(x, y, z, content, color, distance, scale, font, los)
+	local text = {}
+
+	text.position = {x, y, z}
+	text.text = content
+	text.color = color
+	text.distance = distance or 20.0
+	text.scale = scale or 1.0
+	text.font = font or "default"
+	text.los = los or false
+	text.id = uuid()
+	TextsList[#TextsList+1] = text
+
+	return #TextsList
+end
+
+--Edits 3D text specified parameters
+function edit3dText(id, x, y, z, content, color, distance, scale, font, los)
+	
+	local newText = {}
+
+	newText.position = {x, y, z}
+	newText.text = content
+	newText.color = color
+	newText.distance = distance or 20.0
+	newText.scale = scale or 1.0
+	newText.font = font or "default"
+	newText.los = los or false
+	
+	local index = nil
+	for i,v in ipairs(TextsList) do
+		if v.id == id then
+			index = i
+			newText.id = v.id
+		end
+	end
+	if index == nil then
+		return false
+	else
+		TextsList[index] = newText
+		return true	
```

Comment:
Lepiej Texts[id].position i aktualizować po kolei niż zamieniać obiekt

Reply:
wg docsów lua - https://www.lua.org/pil/2.5.html
 When a program has no references to a table left, Lua memory management will eventually delete the table and reuse its memory. 
 
 zakładam że to w tym przypadku też zadziała - stare dane zostaną złapane przez garbage collectora i wywalone?